artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQMessageProducer.java

```diff
@@ -393,57 +395,18 @@ private void doSendx(ActiveMQDestination destination,
          if (defaultDestination == null) {
             throw new UnsupportedOperationException("Destination must be specified on send with an anonymous producer");
          }
-
          destination = defaultDestination;
-      } else {
-         if (defaultDestination != null) {
-            if (!destination.equals(defaultDestination)) {
-               throw new UnsupportedOperationException("Where a default destination is specified " + "for the sender and a destination is " + "specified in the arguments to the send, " + "these destinations must be equal");
-            }
+      } else if (defaultDestination != null) {
+         if (!destination.equals(defaultDestination)) {
+            throw new UnsupportedOperationException("Where a default destination is specified " + "for the sender and a destination is " + "specified in the arguments to the send, " + "these destinations must be equal");
          }
+      } else {
+         session.checkDestination(destination);
 
          address = destination.getSimpleAddress();
```

Comment:
I just spotted i think this should be outside the if else block otherwise address will be null for default

Reply:
Those lines are the reason why I wanted to run a full testsuite before merging this.

And I think you're right...



I'm adding a commit to be squashed into yours. I will squash it before merging it, but if you could review it please

Reply:
back on a proper pc,

i was meaning only:
         address = destination.getSimpleAddress();
 
if defaultdestination != null, then don't need to do session.checkDestination(destination), as we checked the destination already when we created the message producer